rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -1778,6 +1833,26 @@ mod tests {
         assert_eq!(out, vec![Some("1"), Some("2"), Some("3")]);
     }
 
+    #[test]
+    fn test_str_to_str_casts() {
```

Comment:
I recommend two other tests:
1.  The other way as well -- `StringArray` --> `LargeStringArray`
2. Nulls  (maybe `        let a = Arc::new(LargeStringArray::from(vec![Some("foo"), None, Some("ham")])) as ArrayRef;` for example)